packages/field-base/src/active-mixin.js

```diff
@@ -68,7 +50,49 @@ const ActiveMixinImplementation = (superclass) =>
       // the `active` attribute needs to be manually removed from the element.
       // Otherwise, it will preserve on the element until the element is activated once again.
       // The case reproduces for `<vaadin-date-picker>` when closing on `Cancel` or `Today` click.
-      this.removeAttribute('active');
+      this._setActive(false);
+    }
+
+    /**
+     * Sets the `active` attribute on the element if an activation key is pressed.
+     *
+     * @param {KeyboardEvent} event
+     * @protected
+     * @override
+     */
+    _onKeyDown(event) {
+      super._onKeyDown(event);
+
+      if (this.disabled) return;
+
+      if (this._activeKeys.includes(event.key)) {
```

Comment:
nit: maybe we could combine these two `if` statements:

```suggestion
      if (!this.disabled && this._activeKeys.includes(event.key)) {
```